sql/src/test/java/io/crate/analyze/SelectWindowFunctionAnalyzerTest.java

```diff
@@ -140,4 +141,113 @@ public void testOverWithFrameDefinition() {
         assertThat(frameDefinition.start().type(), is(FrameBound.Type.UNBOUNDED_PRECEDING));
         assertThat(frameDefinition.end().type(), is(FrameBound.Type.UNBOUNDED_FOLLOWING));
     }
+
+    @Test
+    public void test_over_references_window_with_order_by() {
+        QueriedSelectRelation<?> analysis = e.analyze(
+            "SELECT AVG(x) OVER w " +
+            "FROM t " +
+            "WINDOW w AS (ORDER BY x)");
+        WindowFunction windowFunction = (WindowFunction) analysis.outputs().get(0);
+        WindowDefinition windowDefinition = windowFunction.windowDefinition();
+        OrderBy orderBy = windowDefinition.orderBy();
+
+        assertThat(orderBy, not(nullValue()));
+        assertThat(orderBy.orderBySymbols().size(), is(1));
+    }
+
+    @Test
+    public void test_over_references_window_with_partition_by() {
+        QueriedSelectRelation<?> analysis = e.analyze(
+            "SELECT AVG(x) OVER w " +
+            "FROM t " +
+            "WINDOW w AS (PARTITION BY x)");
+        WindowFunction windowFunction = (WindowFunction) analysis.outputs().get(0);
+        WindowDefinition windowDefinition = windowFunction.windowDefinition();
+
+        assertThat(windowDefinition.partitions().size(), is(1));
```

Comment:
Personally I prefer  using `contains(isReference("x"))` over the size assertions. It's a more explicit way to assert the expected contents and will give more meaningful information in case the size doesn't match, as it would list the extra contents in the error.

Fine for me to keep it as is, but maybe something to keep in mind for the future.